c++ training/SteamProject/Videogame/Steam.cpp

```diff
@@ -0,0 +1,434 @@
+
+	std::cin.ignore(500, '\n');
+	if (std::cin.peek() == '\n')
+	{
+		OpenMainMenu();
+	}
+}
+//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+void FSteam::OpenCreateCategory()
+{
+	system("cls");
+
+	if (CategoryContainer.IsFull())
+	{
+		std::cout << "There is no available space right now to create another category, please type Enter to return to the previous menu";
+
+		std::cin.ignore(500, '\n');
+		if (std::cin.peek() == '\n')
+		{
+			OpenCategoryManagerMenu();
+		}
+		return;
+	}
+
+	std::cout << "Please enter a name for the new category: ";
+	std::string TemporalCategoryName;
+	std::cin >> TemporalCategoryName;
+
+	std::string CategoryName = ValidateInputAndReturnValidatedInput(TemporalCategoryName, "Please enter a valid name for the new category: ");
+	FCategory Category(CategoryName);
+
+	CategoryContainer.AddCategory(Category);
+
+	system("cls");
+
+	std::cout << "Category created succesfuly. Please type Enter to return to the previous menu";
+
+	std::cin.ignore(500, '\n');
+	if (std::cin.peek() == '\n')
+	{
+		OpenCategoryManagerMenu();
+	}
+	return;
+}
+//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+void FSteam::OpenRemoveCategory()
+{
+	system("cls");
+
+	if (CategoryContainer.IsEmpty())
+	{
+		std::cout << "There is currently no category. Please type Enter to return to the previous menu";
+
+		std::cin.ignore(500, '\n');
+		if (std::cin.peek() == '\n')
+		{
+			OpenCategoryManagerMenu();
+		}
+		return;
+
+	}
+
+	std::cout << "Choose one of the following options or categories to remove:\n" << std::endl << "0 - Don't remove any category, return to the previous menu" << std::endl;
+	for (int i = 1; i <= CategoryContainer.GetCurrentNumberOfCategories(); i++)
+	{
+		std::cout << i << " - " << CategoryContainer.GetCategory(i-1).GetName() << std::endl;
+	}
+
+	std::cout << "\nPlease enter your choice: ";
+
+	float TemporalUserChoice;
+	std::cin >> TemporalUserChoice;
+
+	int UserChoice = ValidateInputAndReturnValidatedInput(TemporalUserChoice, "Please enter a valid choice : ", 0, CategoryContainer.GetCurrentNumberOfCategories());
+
+	if (UserChoice == 0)
+	{
+		OpenCategoryManagerMenu();
+		return;
+	}
+
+	CategoryContainer.RemoveCategory(UserChoice - 1);
+
+	system("cls");
+
+
+	std::cout << "Category removed succesfully. Please type Enter to return to the previous menu ";
+
+	std::cin.ignore(500, '\n');
+	if (std::cin.peek() == '\n')
+	{
+		OpenCategoryManagerMenu();
+	}
+	return;
+}
+//---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
+int FSteam::ChooseCategory()
```

Comment:
This method could be const. (It couldn't currently, but your validation functions could also be const. If those are const, this one can be as well)